edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/DeviceClientWrapper.cs

```diff
@@ -24,9 +24,12 @@ public DeviceClientWrapper(DeviceClient deviceClient)
 
         public Task AbandonAsync(string messageId) => this.underlyingDeviceClient.AbandonAsync(messageId);
 
-        public Task CloseAsync() => this.isActive.GetAndSet(false)
-            ? this.underlyingDeviceClient.CloseAsync()
-            : Task.CompletedTask;
+        public Task CloseAsync()
+        {
+            this.isActive.Set(false);
+            this.underlyingDeviceClient?.Dispose();
```

Comment:
Calling this twice might throw right?
Why not do the isAcive.GetAndSet(false) like the code was?

Reply:
Calling Dispose twice doesn't throw, but I changed back to GetAndSet just to be more safe